### PR TITLE
DTS-40988: applied default threshold on mapping | <Release - 11.1.0>

### PR DESCRIPTION
**Pull request by shunaray**
Change-log: Changes in getter api of field-mapping for threshold to populate

https://publicissapient.atlassian.net/browse/DTS-40988

### CODE REVIEW

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/projectconfig/fieldmapping/service/FieldMappingServiceImpl.java [hunk 1]
```diff
@@ -578,4 +583,42 @@ private void removeTraceLog(ObjectId basicProjectConfigId) {
...
+	 *            the KPI code
+	 * @param field
+	 *            the field name
+	 * @param projectLevelThresholdFields
+	 *            the list of project level threshold fields
+	 * @param value
+	 *            the current value of the field
+	 * @return the populated threshold value, from kpi_master if empty
+	 */
+	private Object populateDefaultKPIThreshold(KPICode kpi, String field, List<String> projectLevelThresholdFields,
```
Comment (ananthpal) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/projectconfig/fieldmapping/service/FieldMappingServiceImpl.java:614]: Minor: populate* methods usually dont have return value. You accept a object and set the values. Appears more like a get method.

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/constant/Constant.java [hunk 2]
```diff
@@ -149,6 +149,7 @@ public final class Constant {
 	public static final String RELIABILITY_REMEDIATION = "reliability_remediation_effort";
 	public static final String DUPLICATED_LINES = "duplicated_lines";
 	public static final String N_CLOC = "ncloc";
+	public static final String PROJECT_LEVEL_THRESHOLD = "Project Level Threshold";
 	private Constant() {
```
Comment (ananthpal) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/constant/Constant.java:153]: Not related to this change. But wondering why was class chosen when interface would have been the best choice.

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/projectconfig/fieldmapping/service/FieldMappingServiceImpl.java (hunk 1 above)
Comment (shunaray) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/projectconfig/fieldmapping/service/FieldMappingServiceImpl.java]: done

**Review by prabasak23 — COMMENTED** [review]
looks good from my side for FE part

**Review by risshukl0 — APPROVED** [review]

**Review by brahmanand1 — APPROVED** [review]

**Review by prabasak23 — APPROVED** [review]
Everything looks fine. Approving.

**Review by Chittauri — APPROVED** [review]

**Pull request merged**